src/availability.rs

```diff
@@ -363,10 +366,107 @@ where
                 Ok(result)
             }
             .boxed()
+        })?
+        .get("get_transaction_range_rev", move |req, state| {
+            async move {
+                let height: usize = req.integer_param("height")?;
+                let offset: usize = req.integer_param("offset")?;
+                let limit: usize = req.integer_param("limit")?;
+                let batch_size = NonZeroUsize::new(10).unwrap();
+
+                let step1 = build_range_stream_rev(height, batch_size);
+                let step2 = step1.flat_map(|range| {
+                    expand_range_to_block_range_stream::<State, Types>(range, state, timeout)
+                });
+
+                let step3 = step2.flat_map(|result| {
+                    futures::stream::iter(match result {
+                        Ok(block) => expand_block_data_to_transaction_vec(block)
+                            .into_iter()
+                            .rev()
+                            .map(Some)
+                            .collect(),
+
+                        _ => vec![None],
+                    })
+                });
+
+                let result = step3
+                    .skip(offset)
+                    .take(limit)
+                    .collect::<Vec<Option<Transaction<Types>>>>()
+                    .await;
+
+                Ok(result)
+            }
+            .boxed()
         })?;
     Ok(api)
 }
 
+// build_range_stream_rev is a helper function that builds a stream of ranges
+// to query for.  It construcs thes streams starting from the end going toward
+// 0.
+fn build_range_stream_rev(
+    end: usize,
+    chunk_size: NonZeroUsize,
+) -> impl Stream<Item = Range<usize>> {
+    let chunk_size = chunk_size.get();
+    let steps = end / chunk_size;
+
+    futures::stream::iter((0..steps).map(move |chunk| {
+        // We start at the back such that step 0 *should* have it's
+        // end being the overall desired end size.
+        let end = end - ((steps - chunk) * chunk_size);
+        let start = max(0, end - chunk_size);
```

Comment:
This `max` doesn't do anything because we're using unsigned integers (as we should be!). `end - chunk_size` won't be negative, it will just underflow, which is very bad